### PR TITLE
Enable testing LCALS in correctness testing

### PR DESCRIPTION
**Pull request by daviditen**
LCALS has only been getting tested for performance testing because with bounds
checks it takes way to long to run. Add a .compopts file with
'--no-bounds-checks' to enable it to run in regular correctness testing.

Modified/added the test config files to run the large loop size for the RAW and
RAW_OMP variants for correctness.

Modified a config param to allow selecting between three output formats:
* MINIMAL: Don't print timing/checksum dumps
* REFERENCE: Like the reference code
* PERF_TEST: Formatted nicely for perf testing

I tried running with bounds checking, and hit an out of bounds error at around
the 40 minute mark. Looking into this, I found that the reference version also
accesses the array out of bounds with index -1.  I added a comment mentioning
this where it happens.

Tested on linux64 and OSX with correctness and performance testing modes.

### CODE REVIEW

File: test/release/examples/benchmarks/lcals/LCALSMain.chpl [hunk 1]
```diff
@@ -160,12 +161,24 @@ module main {
...
 
     writeln("\n generate reports....");
-    if !perfTesting {
-      generateTimingReport(run_variants, output_dirname);
-      generateChecksumReport(run_variants, output_dirname);
-      generateFOMReport(run_variants, output_dirname);
-    } else {
-      generatePerfTimingReport(run_variants);
+
+    select outputFormat {
```
Comment (ronawho) [inline test/release/examples/benchmarks/lcals/LCALSMain.chpl]: Switch outputFormat to an enum instead?

**Comment by ronawho** [comment]
I think it'd be nice to use an enum instead of just different integer sentinels, but looks good other than that!

**Comment by daviditen** [comment]
Thanks! I'll change it.

**Comment by ronawho** [comment]
👍

**Pull request merged**